modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/winback/WinbackViewModel.kt

```diff
@@ -0,0 +1,158 @@
+package au.com.shiftyjelly.pocketcasts.profile.winback
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import au.com.shiftyjelly.pocketcasts.models.type.Subscription
+import au.com.shiftyjelly.pocketcasts.models.type.SubscriptionMapper
+import au.com.shiftyjelly.pocketcasts.models.type.SubscriptionPricingPhase
+import au.com.shiftyjelly.pocketcasts.repositories.subscription.ProductDetailsState
+import au.com.shiftyjelly.pocketcasts.repositories.subscription.PurchasesState
+import au.com.shiftyjelly.pocketcasts.repositories.subscription.SubscriptionManager
+import com.android.billingclient.api.ProductDetails
+import com.android.billingclient.api.Purchase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class WinbackViewModel @Inject constructor(
+    val subscriptionManager: SubscriptionManager,
+) : ViewModel() {
+    private val subscriptionMapper = SubscriptionMapper()
+
+    private val _uiState = MutableStateFlow(UiState.Empty)
+    internal val uiState = _uiState.asStateFlow()
+
+    init {
+        loadInitialPlans()
+    }
+
+    private fun loadInitialPlans() {
+        _uiState.value = _uiState.value.copy(subscriptionPlansState = SubscriptionPlansState.Loading)
```

Comment:
I could be wrong, but it seems like this line isn't needed, as the default is loading.

Reply:
This would assume that the current state is always `Loading` which might not be the case for reloading when added.

Reply:
Oh I see, thanks.